test/test_t2star.py

```diff
@@ -246,19 +241,77 @@ def test_t2star_parallel(self):
 
         for i in range(2):
             sub_res = res.component_experiment_data(i).analysis_result(0)
-            self.assertAlmostEqual(sub_res["t2star_value"], t2star[i], delta=3)
             self.assertAlmostEqual(
-                sub_res["frequency_value"],
-                estimated_freq[i],
-                delta=3,
+                sub_res["t2star_value"], t2star[i] * dt_factor, delta=3 * dt_factor
+            )
+            self.assertAlmostEqual(
+                sub_res["frequency_value"], estimated_freq[i] / dt_factor, delta=3 / dt_factor
             )
-            sub_res = res.component_experiment_data(i).analysis_result(0)
             self.assertEqual(
                 sub_res["quality"],
                 "computer_good",
                 "Result quality bad for experiment on qubit " + str(i),
             )
 
+    def test_t2star_concat_2_experiments(self):
+        """
+        Concatenate the data from 2 separate experiments
+        """
+        unit = "s"
+        dt_factor = 1
+        estimated_t2star = 30
+        estimated_freq = 0.7
+        # First experiment
+        qubit = 0
+        delays1 = list(range(1, 60, 2))
+
+        exp1 = T2StarExperiment(qubit, delays1, unit=unit)
+        default_p0 = {
+            "A": 0.5,
+            "t2star": estimated_t2star,
+            "f": estimated_freq,
+            "phi": 0,
+            "B": 0.5,
+        }
+        exp1.set_analysis_options(user_p0=default_p0)
+        backend = T2starBackend(
+            p0={
+                "a_guess": [0.5],
+                "t2star": [estimated_t2star],
+                "f_guess": [estimated_freq],
+                "phi_guess": [0.0],
+                "b_guess": [0.5],
+            },
+            initial_prob_plus=[0.0],
+            readout0to1=[0.02],
+            readout1to0=[0.02],
+            dt_factor=1,
+        )
+
+        # run circuits
+        expdata1 = exp1.run(backend=backend, shots=1000)
+
+        # second experiment
+        delays2 = list(range(2, 65, 2))
+        exp2 = T2StarExperiment(qubit, delays2, unit=unit)
+        exp2.set_analysis_options(user_p0=default_p0)
+        expdata2 = exp2.run(backend=backend, experiment_data=expdata1, shots=1000)
+        result0 = expdata2.analysis_result(0)
+        result1 = expdata2.analysis_result(1)
```

Comment:
It's a bit confusing that `result0` is the result associated with `expdata1`

Reply:
changed the {1,2} to {0,1}.